pkg/annotations/middleware.go

```diff
@@ -1,50 +1,52 @@
-package provider
+package annotations
 
 import (
 	"errors"
 	"fmt"
 	"strconv"
 
-	"github.com/containous/maesh/pkg/k8s"
-	"github.com/containous/maesh/pkg/topology"
 	"github.com/containous/traefik/v2/pkg/config/dynamic"
 )
 
-// MiddlewareBuilder builds middlewares of a service.
-type MiddlewareBuilder func(svc *topology.Service) (*dynamic.Middleware, error)
+const (
+	annotationRetryAttempts            = baseAnnotation + "retry-attempts"
+	annotationCircuitBreakerExpression = baseAnnotation + "circuit-breaker-expression"
+	annotationRateLimitAverage         = baseAnnotation + "ratelimit-average"
+	annotationRateLimitBurst           = baseAnnotation + "ratelimit-burst"
+)
 
-// Build builds middlewares of the given service using annotations.
-func buildMiddlewareFromAnnotations(svc *topology.Service) (*dynamic.Middleware, error) {
+// BuildMiddleware builds a middleware from the given annotations
+func BuildMiddleware(annotations map[string]string) (*dynamic.Middleware, error) {
 	var middleware dynamic.Middleware
 
 	// Build circuit-breaker middleware.
-	if circuitBreakerExpression, ok := svc.Annotations[k8s.AnnotationCircuitBreakerExpression]; ok {
+	if circuitBreakerExpression, ok := annotations[annotationCircuitBreakerExpression]; ok {
 		middleware.CircuitBreaker = &dynamic.CircuitBreaker{Expression: circuitBreakerExpression}
 	}
 
 	// Build retry middleware.
-	if retryAttempts, ok := svc.Annotations[k8s.AnnotationRetryAttempts]; ok {
+	if retryAttempts, ok := annotations[annotationRetryAttempts]; ok {
 		attempts, err := strconv.Atoi(retryAttempts)
 		if err != nil {
-			return nil, fmt.Errorf("unable to build retry middleware, %q annotation is invalid: %w", k8s.AnnotationRetryAttempts, err)
+			return nil, fmt.Errorf("unable to build retry middleware, %q annotation is invalid: %w", annotationRetryAttempts, err)
 		}
 
 		middleware.Retry = &dynamic.Retry{Attempts: attempts}
 	}
 
 	// Build rate-limit middleware.
-	rateLimitAverage, hasRateLimitAverage := svc.Annotations[k8s.AnnotationRateLimitAverage]
-	rateLimitBurst, hasRateLimitBurst := svc.Annotations[k8s.AnnotationRateLimitBurst]
+	rateLimitAverage, hasRateLimitAverage := annotations[annotationRateLimitAverage]
```

Comment:
Why these annotations don't have a corresponding `get` function?
Same for above annotations.